### ISSUE TITLE
[a11y][Tree] - Tree with Multselect mode - Accessible names not getting announced properly with screen reader

### ISSUE THREAD

**Issue by yoganjan** [SAP/ui5-webcomponents#8073]
### Bug Description

This issue is happening with andrion mobile version . When talkback feature is enabled. 
When user swipe through all the items under -> once click -> it announced as "selected" by talkback which is incorrect. , there is no checkbox visibly present in the screen but screenreader announced as selected.
Also, checkboxes are not associated with its corresponding header text for screenreader users.

### Expected Behaviour

Screen reader should properly announce the text which is visible on the screen. 

### Steps to Reproduce

1. Open https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-tree--docs
2. Select Multi select Mode for the tree component
3. Try selecting the each list item 
4. Observe that the accessible name is not announced by the screen reader properly
5. it announced as "selected" by talkback which is incorrect. , there is no checkbox visibly present in the screen but screenreader announced as selected.
6.
...

### Priority

None

### UI5 Web Components Version

1.20.0

### Browser

Chrome

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by yoganjan** [comment]
Reference Video
https://github.com/SAP/ui5-webcomponents/assets/16238780/d24a221e-669a-42ff-8727-8f79dfb39002

**Comment by kgogov** [comment]
Hello @yoganjan,

I appreciate you bringing this matter to our attention.
I understand that you have concerns with the ui5-tree's accessible names not being announced by the screenreader, specifically on an Android device using Talkback.

However, I need additional information to further understand the issue. As per your message, it appears that the checkboxes are **not visible**. This is unusual because when the **mode** property of the tree is set to `MultiSelect`, the checkboxes **should be visible**.

Kindly refer to the attached screenshot and provide more details - if possible the steps to recreate the problem or any other specific conditions that may be contributing to this issue. If you can, please share an example from [CodeSandbox](https://codesandbox.io/) or a similar platform to better examine the issue.

Best regards,
Konstantin

![Google-Pixel-6-Android-Testing](https://github.com/SAP/ui5-webcomponents/assets/68374332/b9798f7c-c5e1-4940-873c-98bb5338c3dd)
